### ISSUE TITLE
Show exception from likelihood function?

### ISSUE THREAD

**Issue by cdeil** [iminuit/iminuit#294]
@robertazanin and I were having issues debugging why the Python kernel was crashing when running MIGRAD via Gammapy from Jupyter. The problem was that a Python exception occured in the FCN and she was still using iminuit 1.2 .

Still, with iminuit 1.3 I still don't see what exception occurred.

Using this minimal example
```
from iminuit import Minuit

def f(x, y, z):
    1/0

m = Minuit(f, pedantic=False)

m.migrad()
```
I get this:
```
>>> m.migrad()
**************************************************
*                     MIGRAD                     *
**************************************************

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "iminuit/_libiminuit.pyx", line 757, in iminuit._libiminuit.Minuit.migrad (/private/var/folders/t_/_mywtcj146lbk2c99bnxw7z40000gp/T/pip-install-vbxbryr0/iminuit/iminuit/_libiminuit.cpp:9736)
RuntimeError: Exception Occured
fcn is called with following arguments:
    x = +0.000000
    y = +0.000000
    z = +0.000000
```

@HDembinski - Is it possible to either raise, or at least print the exception traceback, to help users debug what is going on?

**Comment by HDembinski** [comment]
Addressed in #297

**Comment by cdeil** [comment]
I still don't see a ZeroDivisionError or a NameError appear in iminuit master, after #297 .

There's also no regression test to show that it works. I'm happy to add that using [pytest.raises](https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions), but I have no idea how to fix the exception handling.

```
>>> from iminuit import Minuit
>>> 
>>> def f(x, y, z):
...     1/0
... 
>>> m = Minuit(f, pedantic=False)
>>> 
>>> m.migrad()
**************************************************
*                     MIGRAD                     *
**************************************************

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "iminuit/_libiminuit.pyx", line 775, in iminuit._libiminuit.Minuit.migrad
    self.cfmin = call_mnapplication_wrapper(
RuntimeError: exception was raised in user function
fcn is called with following arguments:
    x = +0.000000
    y = +0.000000
    z = +0.000000

>>> def f(x):
...     try:
...         1/0
...     except:
...         spam
... 
>>> m = Minuit(f, pedantic=False)
>>> m.migrad()
**************************************************
*                     MIGRAD                     *
**************************************************

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "iminuit/_libiminuit.pyx", line 775, in iminuit._libiminuit.Minuit.migrad
    self.cfmin = call_mnapplication_wrapper(
RuntimeError
```

**Comment by HDembinski** [comment]
Naturally, I tested this on my machine. I don't know how to write a unit test for the traceback. Do you?

**Comment by HDembinski** [comment]
But I agree that this is still broken, I am having another look...

**Comment by cdeil** [comment]
Testing expected exceptions is easy:
https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions
You get an object with the info and can write asserts on that:
> excinfo is a ExceptionInfo instance, which is a wrapper around the actual exception raised. The main attributes of interest are .type, .value and .traceback.

I'm 99% sure I was on latest master. If you add the test I think you'll see the issue in the CI build.

**Comment by HDembinski** [comment]
Thanks for the pointer, I am now using this to check that the error message contains the original information. The problem is actually caused by a bug in Python-3, which you are using. I am developing on Python-2.7, that's why it worked for me.

**Comment by HDembinski** [comment]
@cdeil Could you please confirm that this fixed your issue? And then close this issue? Thank you!

**Comment by cdeil** [comment]
This is what I see now:
```
>>> from iminuit import Minuit
>>> def f(x):
...     try:
...         1 / 0
...     except:
...         spam
... 
>>> m = Minuit(f, pedantic=False)
>>> m.migrad()
**************************************************
*                     MIGRAD                     *
**************************************************

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "iminuit/_libiminuit.pyx", line 775, in iminuit._libiminuit.Minuit.migrad
    self.cfmin = call_mnapplication_wrapper(
RuntimeError: exception was raised in user function
User function arguments:
    x = +0.000000
Original python exception in user function:
NameError: name 'spam' is not defined
  File "<stdin>", line 5, in f
```

@HDembinski - I showing the full chained exception possible without a ton of work / complexity?
If yes, it would be nice to have. If no, let's leave as is and close the issue.

Thanks!

**Comment by HDembinski** [comment]
The ZeroDivisionError is converted into a RuntimeError. All errors that happen inside the user function are converted into a RuntimeErrors. This is a limitation of Cython.
